scilpy/tractanalysis/fibertube_scoring.py

```diff
@@ -437,105 +434,49 @@ def endpoint_connectivity(step_size, blur_radius, centerlines,
         centers, indices, max_seg_length = streamlines_to_segments(
             centerlines, False)
 
-    centerline_fixed_length = len(centerlines[0])-1
+    tree = nbKDTree(centers)
 
-    # Building KDTree with centerline segments
-    kdtree_centers = np.zeros((0, 3))
-    for fi, fiber in enumerate(centerlines):
-        kdtree_centers = np.concatenate(
-            (kdtree_centers, centers[centerline_fixed_length * fi:
-             (centerline_fixed_length * fi + centerlines_length[fi] - 1)]))
-    tree = nbKDTree(kdtree_centers)
+    vc = set()
+    ic = set()
+    nc = set()
 
-    truth_vc = set()
-    truth_ic = set()
-    truth_nc = set()
-    res_vc = set()
-    res_ic = set()
-    res_nc = set()
+    # streamline[1] is the last point with a valid direction
```

Comment:
Maybe add a note somewhere that this must be the output of your fibertube tracking script because I'm not sure this order is always guaranteed. Maybe it's worth a `#todo: add first point index to dps` or something? Then you could use this info to make sure you always have the good first and last point. Also your scoring won't handle seeding from the middle of a fibertube instead of an extremity right?